src/level_gen.cpp

```diff
@@ -119,6 +119,12 @@ static inline Entity createAgent(Engine &ctx, const MapObject &agentInit) {
                 1.5f * math::up);
     }
 
+    ctx.get<PartnerObservations>(agent) = PartnerObservations();
```

Comment:
This won't zero-initialize `PartnerObservations` because `PartnerObservation`'s default constructor doesn't zero-initialize its members.